internal/resources/private_endpoints.go

```diff
@@ -0,0 +1,317 @@
+package resources
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api"
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/errors"
+	providerschema "github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/schema"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &PrivateEndpoint{}
+	_ resource.ResourceWithConfigure   = &PrivateEndpoint{}
+	_ resource.ResourceWithImportState = &PrivateEndpoint{}
+)
+
+type PrivateEndpoint struct {
+	*providerschema.Data
+}
+
+func NewPrivateEndpoint() resource.Resource {
+	return &PrivateEndpoint{}
+}
+
+func (p *PrivateEndpoint) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_private_endpoints"
+}
+
+func (p *PrivateEndpoint) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"organization_id": stringAttribute([]string{required}),
+			"project_id":      stringAttribute([]string{required}),
+			"cluster_id":      stringAttribute([]string{required}),
+			"endpoint_id":     stringAttribute([]string{required}),
+			"status":          stringAttribute([]string{computed}),
+		},
+	}
+}
```

Comment:
Check the scope of adding require Replace

Reply:
it doesn't make sense for a `computed` field